index.js

```diff
@@ -132,6 +132,14 @@ class ApiFactory {
       apiFactory.logger({ level: 'DEBUG', title: 'Original Event, before transformation', originalEvent });
     }
 
+    //transforming event For HttpAPI payload version 2.0
+    if(originalEvent.version === "2.0"){
+      const path = originalEvent.rawPath;
```

Comment:
So I already see handling for exactly this in the `convertEvent` method. (Maybe it makes sense before adding code here, we add a unit test that validates the new event type has the correct transformation.

The only reason I could see that node getting run is if `event.path` doesn't exist in the following `if` block. So if that's the case, updating that check would make more sense. Though I'm not sure that is the only problem. I really think the unit test would be best place to start here.

Reply:
So I could try updating the check for event.path . But I think not all the keys are not getting set in the event object. like  httpMethod, resource. I don't know if they are required to be set.

Reply:
This is why I said start with the unit test, it should be easy to write one, you can use the example from the docs as input and some of the other tests in this library for inspiration.

Let me know if you get stuck with that.